# HG changeset patch
# User Jason Grout <jason.grout@drake.edu>
# Date 1281801383 25200
# Node ID 64ca3b397c20b16c157fefd65c4c87a352e76a31
# Parent  5b338f2e484f2065d3d30d47bc204d6e9ed13d12
#9221: Update the way axes labels are calculated and number of colormaps for matplotlib 1.0

diff -r 5b338f2e484f -r 64ca3b397c20 sage/plot/colors.py
--- a/sage/plot/colors.py	Thu Aug 05 03:35:44 2010 -0700
+++ b/sage/plot/colors.py	Sat Aug 14 08:56:23 2010 -0700
@@ -1230,7 +1230,7 @@
             0
             sage: maps.load_maps()
             sage: len(maps.maps)
-            116
+            134
         """
         global cm
         if not cm:
@@ -1274,7 +1274,7 @@
             sage: from sage.plot.colors import Colormaps
             sage: maps = Colormaps()
             sage: len(maps)
-            116
+            134
         """
         self.load_maps()
         return len(self.maps)
diff -r 5b338f2e484f -r 64ca3b397c20 sage/plot/plot.py
--- a/sage/plot/plot.py	Thu Aug 05 03:35:44 2010 -0700
+++ b/sage/plot/plot.py	Sat Aug 14 08:56:23 2010 -0700
@@ -1861,25 +1861,25 @@
                 # We set the label positions according to where we are
                 # drawing the axes.
                 if xaxis=='bottom':
-                    yaxis_labely=1
+                    yaxis_labely=subplot.get_ylim()[1]
                     yaxis_labeloffset=8
                     yaxis_vert='bottom'
                     xaxis_labely=0
                     xaxis_vert='baseline'
                 else:
-                    yaxis_labely=0
+                    yaxis_labely=subplot.get_ylim()[0]
                     yaxis_labeloffset=-8
                     yaxis_vert='top'
                     xaxis_labely=1
                     xaxis_vert='top'
 
                 if yaxis=='left':
-                    xaxis_labelx=1
+                    xaxis_labelx=subplot.get_xlim()[1]
                     xaxis_labeloffset=8
                     xaxis_horiz='left'
                     yaxis_labelx=0
                 else:
-                    xaxis_labelx=0
+                    xaxis_labelx=subplot.get_xlim()[0]
                     xaxis_labeloffset=-8
                     xaxis_horiz='right'
                     yaxis_labelx=1
